xbmc/cores/VideoRenderers/BaseRenderer.cpp

```diff
@@ -222,11 +222,14 @@ void CBaseRenderer::FindResolutionFromFpsMatch(float fps, float& weight)
 RESOLUTION CBaseRenderer::FindClosestResolution(float fps, float multiplier, RESOLUTION current, float& weight)
 {
   RESOLUTION_INFO curr = g_graphicsContext.GetResInfo(current);
-
+  RESOLUTION_INFO orig = curr;
```

Comment:
assume you start at 1080@60 (desktop res). start some 2160@24 -> switch to higher res. after a while player detects fps of 60, this code won't switch back to 1080. you want to compare to RES_DESKTOP , not current